core/message_pool.go

```diff
@@ -230,12 +263,49 @@ func UpdateMessagePool(ctx context.Context, pool *MessagePool, store *hamt.CborI
 		pool.Remove(cid)
 	}
 
+	// prune all messages that have been in the pool too long
+	return pool.timeoutMessages(ctx, store, new)
+}
+
+// timeoutMessages removes all messages from the pool that arrived more than MessageTimeout tip sets ago.
+// Note that we measure the timeout in the number of tip sets we have received rather than a fixed block
+// height. This prevents us from prematurely timing messages that arrive during long chains of null blocks.
```

Comment:
@acruikshank and I discussed this earlier and I'm in favour. But we want to align this with outbound queue policy, so it would be helpful if the block counting logic were broken out to a `FindAncestor(n int)` function.

Reply:
@phritz I think it is. Fixing the expiry to a block height gives us an ugly tradeoff between making the user wait an unreasonable amount of time for invalidation and the possibility of invalidating invalid messages due to a long chain of null blocks. As @anorth points out, the inbound and outbound queues will be in sync on this. The outbound queue will block new messages until old message clear, so there's no mental math.